web/src/components/onboarding/OnboardingTaxId.tsx

```diff
@@ -100,65 +125,136 @@ export const OnboardingTaxId = ({
     props.onValidation && props.onValidation(fieldName, Object.values(errors(errorMap)).includes(true));
   };
 
-  const error = fieldStates[fieldName].invalid;
+  const setSplitTaxId = (taxId: string) => {
+    setTaxIdValue(taxId.trim().slice(0, 9));
+    setLocationValue(taxId.trim().slice(9, 12));
+  };
 
-  if (initialType.current === "FULL") {
+  const toggleTaxIdDisplay = async () => {
+    if (!state.profileData.taxId) {
+      return;
+    }
+    if (!state.profileData.taxId.includes(maskingCharacter) && !displayTaxId) {
+      if (initialType.current == "SPLIT") {
+        setSplitTaxId(state.profileData.taxId);
+      }
+      setDisplayTaxId(true);
+    } else if (
+      state.profileData.taxId.includes(maskingCharacter) &&
+      !displayTaxId &&
+      state.profileData.encryptedTaxId
+    ) {
+      const decryptedTaxId = await decryptTaxId({ encryptedTaxId: state.profileData.encryptedTaxId });
+      setProfileData({ ...state.profileData, taxId: decryptedTaxId });
+      if (initialType.current == "SPLIT") {
+        setSplitTaxId(decryptedTaxId);
+      }
+      setDisplayTaxId(true);
+    } else {
+      setDisplayTaxId(false);
+    }
+  };
+
+  const hideButtonText = () => {
+    return isTabletAndUp ? Config.tax.hideButtonText : Config.tax.hideButtonTextMobile;
+  };
+
+  const showButtonText = () => {
+    return isTabletAndUp ? Config.tax.showButtonText : Config.tax.showButtonTextMobile;
+  };
+
+  const showHideToggleButton = () => {
     return (
-      <OnboardingField
-        error={error}
-        fieldName={fieldName}
-        visualFilter={formatTaxId}
-        validationText={validationText}
-        numericProps={{ minLength: 12, maxLength: 12 }}
-        handleChange={handleChangeFullTaxId}
-        allowMasking={true}
-        className={props.forTaxTask ? "width-100" : ""}
-        formInputFull={props.formInputFull ? true : false}
-        {...props}
-      />
+      <UnStyledButton style="tertiary" align="start" onClick={toggleTaxIdDisplay}>
+        <ButtonIcon svgFilename={displayTaxId ? "hide" : "show"} />
+        <span className="underline">{displayTaxId ? hideButtonText() : showButtonText()}</span>
+      </UnStyledButton>
     );
-  }
+  };
```

Comment:
stuff like this show/hide toggle could be extracted into its own component, along with extracting its own unit tests. It would make this component AND this component's tests less long and more focused on *its* behavior